crates/modelardb_storage/src/parser.rs

```diff
@@ -47,11 +49,117 @@ use sqlparser::tokenizer::Token;
 use crate::error::{ModelarDbStorageError, Result};
 use crate::metadata::model_table_metadata::{GeneratedColumn, ModelTableMetadata};
 
-/// Constant specifying that a model table should be created.
-pub const CREATE_MODEL_TABLE_ENGINE: &str = "ModelTable";
+/// A top-level statement (SELECT, INSERT, CREATE, UPDATE, etc.) that have been tokenized, parsed,
+/// and for which semantics checks have verified that it is compatible with ModelarDB. CREATE TABLE
+/// and CREATE MODEL TABLE are supported.
+#[derive(Debug)]
+pub enum ModelarDbStatement {
+    /// CREATE TABLE.
+    CreateNormalTable { name: String, schema: Schema },
+    /// CREATE MODEL TABLE.
+    CreateModelTable(Arc<ModelTableMetadata>),
+    /// INSERT, EXPLAIN, SELECT.
+    Statement(Statement),
+    /// INCLUDE addresses SELECT.
+    IncludeSelect(Statement, Vec<String>),
+    /// DROP TABLE.
+    DropTable(Vec<String>),
+    /// TRUNCATE TABLE.
+    TruncateTable(Vec<String>),
+}
+
+/// Tokenize and parse the SQL statements in `sql` and return its parsed representation in the form
```

Comment:
"Tokenize and parse the SQL statements" makes it seem like we support having multiple SQL statements in `sql` even though we return an error in that case.

Reply:
Fixed.